File: problems-labs/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height = get_int("Height: ");
    int spaces = 1;
    int success = 0;

    while (success < 1)
    {
        if (height == 0)
        {
            height = get_int("Height: ");
        }
        if (height >= 1 && height <= 8)
        {
            for (int row = 1; row <= height; row++)
            {
                while (spaces <= (height - row))
                {
                    printf(" ");
                    spaces++;
                }
                spaces = 1;
                for (int hashes = 1; hashes <= row; hashes++)
                {
                    printf("#");
                }
                printf("  ");
                for (int hashes = 1; hashes <= row; hashes++)
                {
                    printf("#");
                }
                if (row <= height)
                {
                    printf("\n");
                }
                spaces = 1;
            }
            success++;
        }
        else
        {
            height = get_int("Height: ");
        }
    }
}

File: problems-labs/sentimental-mario-more/mario.py
# TODO
# Problem: https://cs50.harvard.edu/x/2022/psets/6/mario/more/
from cs50 import get_int

# get user input and set default whitespace
height = get_int("height: ")
spaces = 1

# if invalid input, prompt again
while height > 8 or height < 1:
    height = get_int("height: ")

# for every row, print appropriate spaces and hashes
for i in range(height):
    while spaces < height - i:
        print(" ", end="")
        spaces += 1
    for hashes in range((i) + 1):
        print("#", end="")
    print("  ", end="")
    for hashes in range((i) + 1):
        print("#", end="")
    if i <= height:
        print("")
    spaces = 1
